File: src/main/java/org/devocative/keights/config/CoreDNSProperties.java
package org.devocative.keights.config;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

import java.time.Duration;

@Getter
@Setter
@Component
@ToString
@ConfigurationProperties(prefix = "keights.coredns")
public class CoreDNSProperties {
	private String configMapKey = "Corefile";
	private String configMap = "coredns";
	private String configMapNamespace = "kube-system";
	private Duration rewriteTaskDelay = Duration.ofSeconds(4);
	private RewriteConfig rewriteConfig = new RewriteConfig();

	// ------------------------------

	@Getter
	@Setter
	@ToString
	public static class RewriteConfig {
		private String annotation = "keights.coredns.rewrite/domain-name";
	}
}

File: src/main/kotlin/org/devocative/keights/config/CoreDNSProperties.kt
package org.devocative.keights.config

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.stereotype.Component
import java.time.Duration

@Component
@ConfigurationProperties(prefix = "keights.coredns")
class CoreDNSProperties {
    var configMapKey = "Corefile"
    var configMap = "coredns"
    var configMapNamespace = "kube-system"
    var rewriteTaskDelay: Duration = Duration.ofSeconds(4)
    var rewriteConfig = RewriteConfig()

    // ------------------------------

    class RewriteConfig {
        var annotation = "keights.coredns.rewrite/domain-name"
    }
}
